app/src/components/network/NetworkList.vue

```diff
@@ -53,10 +53,8 @@ export default {
   },
   methods: {
     async selectNetworkHandler(network) {
-      if (this.isExtension) {
-        if (this.networkId !== network.id) {
-          await this.$store.dispatch(`setNetwork`, network)
-        }
+      if (this.networkId !== network.id) {
+        await this.$store.dispatch(`setNetwork`, network)
```

Comment:
so this wasn't ever needed right? This check I mean

Reply:
For me it works fine both ways :thinking: ?

Reply:
can you rephrase?

Reply:
if you are in cosmos-hub-mainnet and create an account for polkadot, the created account is for cosmos

Reply:
but how can you create an account for Polkadot in cosmos-hub-mainnet? You always select the network first, before creating any account? :question:

Reply:
`setNetwork` was not called. But farther down, `currentNetwork` was used for some properties. This variable still had the old network.